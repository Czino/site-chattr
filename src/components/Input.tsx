import { ChangeEvent, InputHTMLAttributes } from 'react'

type Props = InputHTMLAttributes<HTMLInputElement> & {
    onTextChange?: (value: string) => void
}
export const Input = ({ onTextChange, ...props }: Props) => {
    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
        if (onTextChange) onTextChange(event.target.value)
    }
    return (
        <input
            {...props}
            onChange={handleChange}
            className={[
                'resize-none border border-gray-300 rounded-md p-2 flex-shrink-0',
                'focus:outline-none focus:ring-2 focus:ring-blue-500',
            ].join(' ')}
        />
    )
}
