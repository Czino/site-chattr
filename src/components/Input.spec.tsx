import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Input } from './Input'

describe('Input', () => {
    const onTextChange = jest.fn()
    it('should render Input with styles', () => {
        const { asFragment } = render(<Input className="mt-4" value="Text" onTextChange={onTextChange} />)
        expect(asFragment()).toMatchSnapshot()
    })
    it('should call onTextChange handler on text change', () => {
        render(<Input className="mt-4" placeholder="Your message..." onTextChange={onTextChange} />)
        userEvent.type(screen.getByPlaceholderText('Your message...'), 'a message')
        expect(onTextChange).toHaveBeenCalledWith('a message')
    })
})
