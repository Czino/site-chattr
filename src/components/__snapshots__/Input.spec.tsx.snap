// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input with styles 1`] = `
<DocumentFragment>
  <input
    class="resize-none border border-gray-300 rounded-md p-2 flex-shrink-0 focus:outline-none focus:ring-2 focus:ring-blue-500"
    value="Text"
  />
</DocumentFragment>
`;
